samples/net/sockets/echo/README.rst

```diff
@@ -1,7 +1,8 @@
-.. _sockets-echo-sample:
+.. zephyr:code-sample:: socket-echo
+   :name: Echo server
```

Comment:
done -- FWIW maybe we could also go for "Simple TCP echo server" but then it would not show up next to the other two when sorted alphabetically. Or "Echo server (TCP only)" is another option I guess. Probably overthinking here :)